lib/commands/interactive-cloud-command.ts

```diff
@@ -0,0 +1,44 @@
+import { isInteractive } from "../helpers";
+import { CloudOperationMessageTypes, CloudCommunicationEvents } from "../constants";
+
+export abstract class InteractiveCloudCommand implements ICommand {
+	public allowedParameters: ICommandParameter[];
+
+	constructor(private interactiveService: ICloudOperationService,
+		protected $errors: IErrors,
+		protected $logger: ILogger,
+		protected $prompter: IPrompter) { }
+
+	public async execute(args: string[]): Promise<void> {
+		this.interactiveService.on(CloudCommunicationEvents.MESSAGE, async (msg: ICloudOperationMessage<any>) => {
```

Comment:
Not sure if this is applicable, as commands in CLI are not long-running and I guess the process will exit once the command completes. But I think we do not remove this handler anywhere

Reply:
It won't hurt to remove it on precess exit.